### PR TITLE
lightning: fix duplicate check panic

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?
Fix duplicate iterator panic when lightning restart from import engine.

### What is changed and how it works?

- Set some missging field in `CloseEngine` to avoid nil pointer panic
- Use RestoreController.sysvars to build KvDecoder in duplicate manager
- Use a new context to stop set service GC safepoint
- Fix a typo in doc

Tests 

- [ ] Unit test
- [x] Integration test 
  - several integration tests will panic before this pr.
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by kennytm — APPROVED** [review]

**Review by gozssky — APPROVED** [review]

File: br/pkg/lightning/backend/local/local.go [hunk 1]
```diff
@@ -903,6 +903,7 @@ func NewLocalBackend(
 	g glue.Glue,
 	maxOpenFiles int,
 	errorMgr *errormanager.ErrorManager,
+	sysVars map[string]string,
```
Comment (gozssky) [inline br/pkg/lightning/backend/local/local.go:906]: It seems `sysVars` is unsued.

**Review by gozssky — COMMENTED** [review]
rest LGTM

**Pull request merged**